### PR TITLE
TAXII2 - Report obj issues

### PR DESCRIPTION
**Pull request by MLainer1**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://jira-hq.paloaltonetworks.local/browse/CIAC-8141 https://jira-hq.paloaltonetworks.local/browse/CIAC-8190

## Description
A few sentences describing the overall goals of the pull request's commits.

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/TAXIIServer/Integrations/TAXII2Server/TAXII2Server_test.py [hunk 1]
```diff
@@ -719,3 +719,45 @@ def test_reports_objects_with_relationships(mocker, taxii2_server_v21):
...
+    mocker.patch.object(demisto, 'searchIndicators', return_value=iocs)
+    create_entity_b_stix_objects(relationships, ioc_value_to_id, [])
+
+    assert None not in ioc_value_to_id
+
+
+def test_create_entity_b_stix_objects_with_revoked_relationship(mocker, taxii2_server_v21):
+    """
+        Given
+            Reports object with relationships
```
Comment (moishce) [inline Packs/TAXIIServer/Integrations/TAXII2Server/TAXII2Server_test.py:748]:
```suggestion
            Reports object with revoked relationships
```

File: Packs/TAXIIServer/Integrations/TAXII2Server/TAXII2Server_test.py [hunk 2]
```diff
@@ -719,3 +719,45 @@ def test_reports_objects_with_relationships(mocker, taxii2_server_v21):
...
+        When
+            Calling handle_report_relationships.
+        Then
+            Validate that each report contained its relationship in the object_refs.
```
Comment (moishce) [inline Packs/TAXIIServer/Integrations/TAXII2Server/TAXII2Server_test.py:752]:
```suggestion
            Validate that the report not contained the revoked relationship in the object_refs.
```

**Review by moishce — COMMENTED** [review]
Looks good

File: Packs/TAXIIServer/ReleaseNotes/2_0_45.md [hunk 3]
```diff
@@ -0,0 +1,7 @@
+
+#### Integrations
+
+##### TAXII2 Server
+
+- Fixed an issue where the entity b of a relationship was returned when the relationship was revoked.
+- Fixed an issue where the relationship ref was missing when returning a relationship between a Report object and a File object.
```
Comment (dantavori) [inline Packs/TAXIIServer/ReleaseNotes/2_0_45.md:7]:
```suggestion
- Fixed an issue where after revoking a relationship the related indicator was still retrieved as part of a collection.
- Fixed an issue where the relationship reference was missing when returning a relationship between a report and a file.
```

**Review by dantavori — APPROVED** [review]

**Comment by MLainer1** [comment]
This PR needs a force merge due to a failure:
image

Opened a ticket to address the issue: https://jira-hq.paloaltonetworks.local/browse/CIAC-8335

**Pull request merged**